tekton/integration_pipeline.go

```diff
@@ -179,6 +180,22 @@ func (r *IntegrationPipelineRun) WithIntegrationLabels(integrationTestScenario *
 	return r
 }
 
+// WithIntegrationAnnotations copies the App Studio annotations from the
+// IntegrationTestScenario to the PipelineRun
+func (r *IntegrationPipelineRun) WithIntegrationAnnotations(its *v1beta1.IntegrationTestScenario) *IntegrationPipelineRun {
+	if r.ObjectMeta.Annotations == nil {
```

Comment:
Here we can use AddAnnotations from operator-toolkit, it handles all special cases

Reply:
Can't use [`AddAnnotations`](https://pkg.go.dev/github.com/redhat-appstudio/operator-toolkit/metadata#AddAnnotations) this function filters to copy `*appstudio.openshift.io/*` annotations, i.e. it doesn't copy all annotations. I've added a call to [`SetAnnotation`](https://pkg.go.dev/github.com/redhat-appstudio/operator-toolkit/metadata#SetAnnotation) though.